src/v/rpc/connection_cache.h

```diff
@@ -34,11 +34,11 @@ class connection_cache final
     using underlying = std::unordered_map<model::node_id, transport_ptr>;
     using iterator = typename underlying::iterator;
 
-    static inline ss::shard_id shard_for(
+    ss::shard_id shard_for(
```

Comment:
Seems like a good time to add a comment to this function, especially given the ambiguity of some of the parameter names.

As I understand it, is it given that we are currently on (self, shard) with max_shards total shards, we would like to know the shard id (return value) we should use to connect to node node.